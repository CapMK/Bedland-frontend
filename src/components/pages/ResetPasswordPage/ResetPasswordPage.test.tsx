import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import PasswordResetPage from './ResetPasswordPage';

describe('PasswordReset Page', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <BrowserRouter>
            <PasswordResetPage />
          </BrowserRouter>
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
