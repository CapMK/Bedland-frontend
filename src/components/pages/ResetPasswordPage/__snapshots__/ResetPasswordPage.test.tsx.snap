// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordReset Page > renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 17.625rem;
  width: 22.5rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 11.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #0A1122;
  width: 100%;
}

.c4 {
  padding: 0rem 2px 0rem 2.9375rem;
  width: 100%;
}

.c5 {
  liststyletype: disc;
  margin-left: 10px;
  padding: 0 0 0 1rem;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  width: 100%;
}

.c9 {
  border: 1px solid #919AFF;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c9:disabled,
.c9:hover,
.c9:active,
.c9:focus {
  outline: none;
}

.c8 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.6875rem;
  width: 19.875rem;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 0.5rem;
  color: #FF7C33;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c11:hover {
  background-color: #FF7C330D;
  border: 1px solid transparent;
}

.c12 {
  background-color: transparent;
  border: 1.75px solid transparent;
  color: #FF7C33;
  padding: undefinedrem 1rem;
}

.c12:hover {
  background-color: #FF7C330D;
  border: 1.75px solid transparent;
}

.c12:focus {
  background-color: #FF7C3326;
  border: 1.75px solid transparent;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  padding: 0 2.84375rem 0rem 2.84375rem;
  width: 100%;
}

.c2 {
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
  height: 27.6875rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25.625rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  <img
    alt=""
    className="c1"
    src="/src/assets/img/LogoResident.svg"
  />
  <div
    className="c2"
  >
    <h1
      className="c3"
    >
      Reset your password
    </h1>
    <div
      className="c4"
    >
      <span>
         Remember, your password must include:
      </span>
       
      <ol
        className="c5"
        style={
          {
            "listStyleType": "disc",
          }
        }
      >
        <li>
          8-24 characters
        </li>
        <li>
          At least one uppercase character
        </li>
        <li>
          At least one special character
        </li>
      </ol>
    </div>
    <form
      className="c6"
      onSubmit={[Function]}
    >
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="password"
          size="medium"
        >
          Your new password
        </label>
        <input
          className="c9"
          data-testid="password"
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter your new password"
          type="password"
        />
      </div>
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="confirmPassword"
          size="medium"
        >
          Confirm the password
        </label>
        <input
          className="c9"
          data-testid="confirmPassword"
          id="confirmPassword"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Confirm the password"
          type="password"
        />
      </div>
      <span
        className="c10"
      />
      <button
        className="c11 c12"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="submit"
      >
        Reset my password
      </button>
    </form>
  </div>
</div>
`;
